server_test.go

```diff
@@ -1,6 +1,112 @@
 package main
 
-import "testing"
+import (
+	"testing"
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"net"
+	"time"
+	"github.com/stretchr/testify/require"
```

Comment:
Not a fan of introducing a _new_ dependency for something simple like performing assertion checks. Historically, we've avoided using such packages in favor of context specific unit test utility functions.

Reply:
Would you prefer for me to reimplement require.Equals(...) and the other functions within a common test file within LND? I'm not terribly opposed to that. I strongly prefer having single line assertion checks over writing out an "if" with a "t.Fatalf" though.

The benefits of single line assertions in unit tests are
1. Having to write less lines of code to implement a check incentivizes writing more checks.
2. Automatically handle presenting the expected vs actual values, whereas people would normally have to manually format the error string with those values, ensuring that they rarely actually do that. Non-descriptive error messages make for worse unit tests. 
3. Standardized error messaging.

Note that the assertions provided by testify are general purpose. For example my previous unit test PR for commands.go uses exclusively the checks provided by testify for making assertions.

Again I'm happy to write the equivalent assertion helper functions within lnd itself if need be to avoid adding another dependency. I'd just like to have some way of decreasing the effort involved in the already tedious task of properly unit testing existing code.